packages/arb-util/broadcaster/clientmanager.go

```diff
@@ -95,25 +107,44 @@ func (cm *ClientManager) RemoveAll() {
 		i++
 	}
 
-	// Only called by destructor, so keep mutex while looping through client list
-	for i := range clientList {
-		cm.remove(clientList[i])
+	// Only called after main ClientManager thread exits, so use removeClient directly
+	for _, client := range clientList {
+		cm.removeClient(client)
 	}
 }
 
-// Remove removes client from stream.
-func (cm *ClientManager) Remove(clientConnection *ClientConnection) {
-	cm.mu.Lock()
-	defer cm.mu.Unlock()
+func (cm *ClientManager) removeClient(clientConnection *ClientConnection) bool {
+	if !cm.clientPtrMap[clientConnection] {
+		return false
+	}
+
+	clientConnection.Stop()
+
+	err := cm.poller.Stop(clientConnection.desc)
+	if err != nil {
+		logger.Warn().Err(err).Msg("Failed to stop poller")
+		return false
+	}
+
+	err = clientConnection.conn.Close()
+	if err != nil {
+		logger.Warn().Err(err).Msg("Failed to close client connection")
+	}
+
+	delete(cm.clientPtrMap, clientConnection)
 
-	cm.remove(clientConnection)
+	// TODO: properly close file descriptor
+	//_ = clientConnection.desc.Close()
```

Comment:
I know this was just moved around, but what's the status of this TODO?

Reply:
It was a reminder to confirm everything was properly closed.  I have confirmed that, and removed the todo.